src/BDCS/API/V0.hs

```diff
@@ -1629,6 +1639,47 @@ compose cfg@ServerConfig{..} ComposeBody{..} test | cbType `notElem` supportedOu
             RecipesDepsolveResponse [] errs    -> return $ ComposeResponse False [] (map (T.pack . show) errs)
             RecipesDepsolveResponse (deps:_) _ -> fn deps
 
+    processCustomization :: RecipeCustomization -> IO [Customization]
+    processCustomization RecipeCustomization{..} = do
+        currentTime <- utcTimeToPOSIXSeconds <$> getCurrentTime
+        let hostnameCustomization = maybe [] (\h -> [processHostname currentTime h]) rcHostName
+            sshKeyCustomizations  = concatMap (processSshKey currentTime) rcSshKeys
+        return $ hostnameCustomization ++ sshKeyCustomizations
+
+    processHostname :: POSIXTime -> String -> Customization
+    processHostname currentTime hostname =
+        let hostnameData = C8.pack $ hostname ++ "\n"
+            hostnameFile = Files "/etc/hostname"
+                                 "root"
+                                 "root"
+                                 (floor currentTime)
+                                 Nothing
+                                 (fromIntegral $ regularFileMode .|. 0o0644)
+                                 (C8.length hostnameData)
+                                 Nothing
+         in WriteFile hostnameFile (Just hostnameData)
+
+    processSshKey :: POSIXTime -> RecipeSshKey -> [Customization]
+    processSshKey currentTime RecipeSshKey{..} =
+        let keyData = C8.pack $ rcSshKey ++ "\n"
+            keyDir  = Files (T.pack $ "/home" </> rcSshUser </> ".ssh")
+                            (T.pack rcSshUser)
+                            (T.pack rcSshUser)
+                            (floor currentTime)
+                            Nothing
+                            (fromIntegral $ directoryMode .|. 0o0700)
+                            0
+                            Nothing
+            keyFile = Files (T.pack $ "/home" </> rcSshUser </> ".ssh" </> "authorized_keys")
+                            (T.pack rcSshUser)
+                            (T.pack rcSshUser)
+                            (floor currentTime)
+                            Nothing
+                            (fromIntegral $ regularFileMode .|. 0o0644)
+                            (C8.length keyData)
+                            Nothing
+         in [WriteFile keyDir Nothing, WriteFile keyFile (Just keyData)]
+
```

Comment:
It might make sense to put these into a separate file.  I imagine the set of customizations is going to grow, potentially to be pretty large, and this file is already unwieldy.